src/librustc_resolve/macros.rs

```diff
@@ -385,10 +385,9 @@ impl<'a> Resolver<'a> {
         if let Some(suggestion) = find_best_match_for_name(self.macro_names.iter(), name, None) {
             if suggestion != name {
                 err.help(&format!("did you mean `{}!`?", suggestion));
-            } else {
-                err.help(&format!("have you added the `#[macro_use]` on the module/import?"));
             }
         }
+        err.help(&format!("have you added the `#[macro_use]` on the module/import?"));
```

Comment:
I think this should say "have you added the `#[macro_use]` on the `extern crate`", but only if there is a macro with that name in an extern crate.

You should be able to use `resolver.lookup_import_candidates()` (in `librustc_resolve/lib.rs`) with a `filter_fn` checks for `Def::Macro`s to check if there are any relevant macros in `extern crate`s that the user might want to `#[macro_use]`.